slicepool/slicepool.go

```diff
@@ -0,0 +1,85 @@
+// Copyright 2020 Jigsaw Operations LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package slicepool
+
+import (
+	"sync"
+)
+
+// Pool wraps a sync.Pool of *[]byte.  To encourage correct usage,
+// all public methods are on slicepool.Slice.
+//
+// All copies of a Pool refer to the same underlying pool.
+//
+// "*[]byte" is used to avoid an allocation due to casting
+// []byte to interface{} when calling sync.Pool.Put.
+type Pool struct {
+	pool *sync.Pool
+	len  int
+}
+
+// MakePool returns a Pool of slices with the specified length.
+func MakePool(sliceLen int) Pool {
+	return Pool{
+		pool: &sync.Pool{
+			New: func() interface{} {
+				slice := make([]byte, sliceLen)
+				return &slice
```

Comment:
This seems odd. You're taking the address of the local variable? Won't `slice` go out of scope?
Shouldn't you `return slice`?

Reply:
Thanks to garbage collection, returning a pointer to a local variable works fine.  The weird construction is because if the heap stores a `[]byte` directly, then every call to pool.Get() returns a copy of the [slice-struct](https://medium.com/rungo/the-anatomy-of-slices-in-go-6450e3bb2b94), and every call to pool.Put() leaks that new slice-struct to the heap.

Reply:
Perhaps add a comment on why we use a pointer, so it doesn't get reverted in the future.

Reply:
Done